crates/apollo-compiler/src/schema/from_ast.rs

```diff
@@ -125,10 +133,14 @@ impl SchemaBuilder {
                         Entry::Occupied(entry) => {
                             let previous = entry.get();
                             if $is_scalar && previous.is_built_in() {
-                                self.errors.push(
-                                    $def.location(),
-                                    BuildError::BuiltInScalarTypeRedefinition,
-                                )
+                                if self.allow_builtin_redefinitions {
+                                    continue;
```

Comment:
Skipping them means that, when you write:

```graphql
"All strings in our API are ASCII-only."
scalar String @metadata(argument: "value")
```

you can not then access the custom description and directives:

```rust
let s = schema.types.get("String").unwrap();
assert!(s.directives().get("metadata").is_some());
```

which I think is surprising.

How about replacing the type with the custom one? It would make things a bit trickier because we'd also want to re-serialize custom scalars that override builtin ones, I guess. (because unlike graphql-js, apollo-rs cares about directives)

(Another approach to solve the confusion could just be to call the option `ignore_builtin_redefinitions` 😄 .)

Reply:
renamed `allow_builtin_redefinitions` -> `ignore_builtin_redefinitions` in https://github.com/apollographql/apollo-rs/pull/990/commits/25716eb5276f73da032f4f53c927c34589174892